### PR TITLE
Add trial text to bold and add underline

### PR DESCRIPTION
**Pull request by deeppandya**
Resolves https://github.com/brave/brave-browser/issues/35913

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by tapanmodh — APPROVED** [review]
lgtm

File: browser/ui/android/strings/android_brave_strings.grd [hunk 1]
```diff
@@ -2398,7 +2398,10 @@ Are you sure you want to do this?
       Protect 5 devices on 1 subscription
     </message>
     <message name="IDS_TRIAL_TEXT" desc="Brave VPN plan text">
-      All plans include a free 7-day trial!
+      All plans include a <ph name="TRIAL_DAYS">%1$s</ph>
+    </message>
+    <message name="IDS_DAY_7_TRIAL_TEXT" desc="Brave VPN plan text">
+      free 7-day trial!
```
Comment (mkarolin) [inline browser/ui/android/strings/android_brave_strings.grd:2404]: Is there no way of doing the underlining without splitting the text into 2 strings? Something along the lines of https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/search_engines/android/java/src/org/chromium/chrome/browser/search_engines/SogouPromoDialog.java;l=109?

Comment (deeppandya) [reply browser/ui/android/strings/android_brave_strings.grd:2404]: @mkarolin we need to apply underline for partial text. so we need to get the start and end of the character.

Comment (mkarolin) [reply browser/ui/android/strings/android_brave_strings.grd:2404]: Bummer, it's not great for l10n to split strings like that.

**Review by mkarolin — APPROVED** [review]
`strings`++

**Pull request merged**